twitter-eventlistener-plugin/src/main/java/com/twitter/presto/plugin/eventlistener/Stringify.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.twitter.presto.plugin.eventlistener;
+
+import com.facebook.presto.spi.eventlistener.QueryCompletedEvent;
+import com.facebook.presto.spi.eventlistener.QueryContext;
+import com.facebook.presto.spi.eventlistener.QueryCreatedEvent;
+import com.facebook.presto.spi.eventlistener.QueryFailureInfo;
+import com.facebook.presto.spi.eventlistener.QueryMetadata;
+import com.facebook.presto.spi.eventlistener.QueryStatistics;
+import com.facebook.presto.spi.eventlistener.SplitCompletedEvent;
+import com.facebook.presto.spi.eventlistener.SplitFailureInfo;
+import com.facebook.presto.spi.eventlistener.SplitStatistics;
+
+import java.time.Duration;
+import java.util.Optional;
+
+class Stringify
+{
+  private static final String DASH = "-";
+  private static final String FIELD_DELIMITER = " ,";
+  private static final String OBJECT_DELIMITER = "\n\t";
+
+  private static String getMillisOrElse(Optional<Duration> d, String defaultVal)
+  {
+    return d.isPresent() ? String.valueOf(d.get().toMillis()) : defaultVal;
+  }
+
+  private Stringify()
+  {}
+
+  public static String toString(QueryContext o)
+  {
+    String[] content = {"environment(%s)",
+                        "user(%s)",
+                        "userAgent(%s)",
+                        "source(%s)",
+                        "catalog(%s)",
+                        "schema(%s)",
+                        "principal(%s)",
+                        "remoteClientAddress(%s)"};
+    return String.format(
+      String.format("QueryContext: %s", String.join(FIELD_DELIMITER, content)),
```

Comment:
By dynamically generating the String that's passed to String.format, we forgo the compile time checks that the format of the string matches the type and number of args passed, which is a common bug. I'd recommend not using an array to generate it dynamically and just inline it. It doesn't seem worth it just to share a comma as a delimiter.